src/arch/pmp.rs

```diff
@@ -81,14 +115,79 @@ pub struct PmpGroup {
 pub struct PmpFlush();
 
 impl PmpGroup {
-    pub const fn new(nb_pmp: usize) -> Self {
+    const fn new(nb_pmp: usize) -> Self {
         PmpGroup {
             pmpaddr: [0; 64],
             pmpcfg: [0; 8],
+            modified: false,
             nb_pmp: nb_pmp as u8,
+            nb_virt_pmp: 0,
         }
     }
 
+    pub fn init_pmp_group(nb_pmp: usize) -> PmpGroup {
+        let mut pmp = Self::new(nb_pmp);
+        let virtual_devices = Plat::create_virtual_devices();
+
+        // Configure PMP registers, if available
+        if pmp.nb_pmp >= 8 {
+            // By activating this entry it's possible to catch all memory accesses
+            pmp.set(ALL_CATCH_OFFSET, usize::MAX, pmpcfg::INACTIVE);
+
+            // Protect Miralis
+            let (start, size) = Plat::get_miralis_memory_start_and_size();
+            pmp.set(
+                MIRALIS_OFFSET,
+                build_napot(start, size).unwrap(),
+                pmpcfg::NAPOT,
+            );
+
+            // Protect virtual devices
+            pmp.set(
+                DEVICES_OFFSET,
+                build_napot(virtual_devices[0].start_addr, virtual_devices[0].size).unwrap(),
+                pmpcfg::NAPOT,
+            );
+
+            pmp.set(
+                DEVICES_OFFSET + 1,
+                build_napot(virtual_devices[1].start_addr, virtual_devices[1].size).unwrap(),
+                pmpcfg::NAPOT,
+            );
+
+            // This PMP entry is used by the policy module for its own purpose
+            #[allow(clippy::reversed_empty_ranges)]
+            for idx in 0..POLICY_SIZE {
+                pmp.set(POLICY_OFFSET + idx, 0, pmpcfg::INACTIVE);
+            }
+
+            // Add an inactive 0 entry so that the next PMP sees 0 with TOR configuration
+            pmp.set(INACTIVE_ENTRY_OFFSET, 0, pmpcfg::INACTIVE);
+
+            // Finally, set the last PMP to grant access to the whole memory
+            pmp.set(
+                (pmp.nb_pmp - 1) as usize,
+                usize::MAX,
+                pmpcfg::RWX | pmpcfg::NAPOT,
+            );
+
+            // Compute the number of virtual PMPs available
+            // Let's see if you are a good german reviewer
```

Comment:
@CharlyCst, je t'ai eu!

Reply:
Really? I actually saw it and added a comment, but indeed I can't find it in the review, I guess I miss-clicked...